packages/sync/src/replicastore/class-table-checksum.php

```diff
@@ -0,0 +1,386 @@
+use Exception;
+use WP_error;
+
+// TODO add rest endpoints to work with this, hopefully in the same folder
+
+class Table_Checksum {
+	public $table                 = '';
+	public $table_configuration   = array();
+	public $range_field           = '';
+	public $key_fields            = array();
+	public $checksum_fields       = array();
+	public $additional_filter_sql = '';
+
+
+	public $default_tables = array();
+
+	public $salt = '';
+
+	public $allowed_tables = array();
+
+	/**
+	 * Table_Checksum constructor.
+	 *
+	 * @param string $table
+	 * @param string $salt
+	 * @param string $range_field
+	 * @param null   $key_fields
+	 * @param null   $filter_field
+	 * @param array  $checksum_fields
+	 */
+
+	public function __construct( $table, $salt = null ) {
+		$this->salt = $salt;
+
+		global $wpdb;
+
+		$this->default_tables = array(
+			'posts'              => array(
+				'table'           => $wpdb->posts,
+				'range_field'     => 'ID',
+				'key_fields'      => array( 'ID' ),
+				'checksum_fields' => array( 'post_modified' ),
+				'filter_sql'      => Settings::get_blacklisted_post_types_sql(),
+			),
+			'postmeta'           => array(
+				'table'           => $wpdb->postmeta,
+				'range_field'     => 'post_id',
+				'key_fields'      => array( 'post_id', 'meta_key' ),
+				'checksum_fields' => array( 'meta_key', 'meta_value' ),
+				'filter_sql'      => Settings::get_whitelisted_post_meta_sql(),
+			),
+			'comments'           => array(
+				'table'           => $wpdb->comments,
+				'range_field'     => 'comment_ID',
+				'key_fields'      => array( 'comment_ID' ),
+				'checksum_fields' => array( 'comment_content' ),
+				'filter_sql'      => Settings::get_comments_filter_sql(),
+			),
+			'commentmeta'        => array(
+				'table'           => $wpdb->commentmeta,
+				'range_field'     => 'comment_id',
+				'key_fields'      => array( 'comment_id', 'meta_key' ),
+				'checksum_fields' => array( 'meta_key', 'meta_value' ),
+				'filter_sql'      => Settings::get_whitelisted_comment_meta_sql(),
+			),
+			'terms'              => array(
+				'table'           => $wpdb->terms,
+				'range_field'     => 'term_id',
+				'key_fields'      => array( 'term_id' ),
+				'checksum_fields' => array( 'term_id', 'name', 'slug' ),
+			),
+			'termmeta'           => array(
+				'table'           => $wpdb->termmeta,
+				'range_field'     => 'term_id',
+				'key_fields'      => array( 'term_id', 'meta_key' ),
+				'checksum_fields' => array( 'meta_key', 'meta_value' ),
+			),
+			'term_relationships' => array(
+				'table'           => $wpdb->term_relationships,
+				'range_field'     => 'object_id',
+				'key_fields'      => array( 'object_id' ),
+				'checksum_fields' => array( /** TODO */ ),
+			),
+			'term_taxonomy'      => array(
+				'table'           => $wpdb->term_taxonomy,
+				'range_field'     => 'term_taxonomy_id',
+				'key_fields'      => array( 'term_taxonomy_id' ),
+				'checksum_fields' => array( /** TODO */ ),
```

Comment:
We have had a lot of data irregularities with this table. From global terms to not updating/removing old records during bulk actions or Full Syncs.

I'd recommend :
term_taxonomy_id, term_id, taxonomy, description and parent 
(basically a whole table minus count which isn't needed for data integrity)

Reply:
Added in 4393a868e